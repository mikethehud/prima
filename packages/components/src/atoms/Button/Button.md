#### Normal

```jsx
<Button text="Hello world" />{' '}
<Button text="Hello world" color="primary" />{' '}
<Button text="Hello world" color="secondary" />{' '}
<Button text="Hello world" color="success" />{' '}
<Button text="Hello world" color="attention" />
```

#### Outline

```jsx
<Button text="Hello world" type="outline" />
<Button text="Hello world" type="outline" color="primary" />
<Button text="Hello world" type="outline" color="secondary" />
<Button text="Hello world" type="outline" color="success" />
<Button text="Hello world" type="outline" color="attention" />
```

#### Ghost

```jsx
<Button text="Hello world" type="ghost" />
<Button text="Hello world" type="ghost" color="primary" />
<Button text="Hello world" type="ghost" color="secondary" />
<Button text="Hello world" type="ghost" color="success" />
<Button text="Hello world" type="ghost" color="attention" />
```

#### Arrow

A button can have an arrow.

```jsx
<Button text="Hello world" color="primary" arrow="true" />
<Button text="Hello world" color="primary" arrow="true" type="outline" />
<Button text="Hello world" color="primary" arrow="true" type="ghost" />
```
