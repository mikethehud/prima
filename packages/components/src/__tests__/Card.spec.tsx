/* tslint:disable:jsx-no-lambda */

import { mount, shallow } from "enzyme"
import * as React from "react"
import Button from "../atoms/Button/Button"
import {
  Card,
  CardBody,
  CardButtons,
  CardSubtitle,
  CardTitle
} from "../molecules/Card/Card"
import { ThemeColor } from "../types"

const highlightColors: ThemeColor[] = [
  "default",
  "primary",
  "secondary",
  "success",
  "attention",
  "negative"
]

describe("Card", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Card />)
    expect(wrapper).toMatchSnapshot()
  })

  // Run through all highlightColors
  highlightColors.forEach(highlightColor => {
    it(`can have highlight with color color ${highlightColor}`, () => {
      const wrapper = shallow(<Button text="Hello" color={highlightColor} />)
      expect(wrapper.hasClass(highlightColor)).toBe(true)
      expect(wrapper).toMatchSnapshot()
    })
  })

  it("renders a cardbody", () => {
    const wrapper = mount(
      <Card>
        <CardBody>Hello</CardBody>
      </Card>
    )
    expect(wrapper.find(".card-body").length).toEqual(1)
    expect(wrapper).toMatchSnapshot()
  })

  it("renders cardbuttons", () => {
    const wrapper = mount(
      <Card>
        <CardButtons>
          <Button text="Hello" />
        </CardButtons>
      </Card>
    )
    expect(wrapper.find(".card-buttons").length).toEqual(1)
    expect(wrapper).toMatchSnapshot()
  })

  it("renders subtitle", () => {
    const wrapper = mount(
      <Card>
        <CardSubtitle>Hello</CardSubtitle>
      </Card>
    )
    expect(wrapper.find(".card-subtitle").length).toEqual(1)
    expect(wrapper).toMatchSnapshot()
  })

  it("renders title", () => {
    const wrapper = mount(
      <Card>
        <CardTitle>Hello</CardTitle>
      </Card>
    )
    expect(wrapper.find(".card-title").length).toEqual(1)
    expect(wrapper).toMatchSnapshot()
  })
})
