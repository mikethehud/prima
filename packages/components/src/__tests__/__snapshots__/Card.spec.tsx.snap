// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card can have highlight with color color attention 1`] = `
<button
  className="button normal attention"
>
  <span>
    Hello
  </span>
</button>
`;

exports[`Card can have highlight with color color default 1`] = `
<button
  className="button normal default"
>
  <span>
    Hello
  </span>
</button>
`;

exports[`Card can have highlight with color color negative 1`] = `
<button
  className="button normal negative"
>
  <span>
    Hello
  </span>
</button>
`;

exports[`Card can have highlight with color color primary 1`] = `
<button
  className="button normal primary"
>
  <span>
    Hello
  </span>
</button>
`;

exports[`Card can have highlight with color color secondary 1`] = `
<button
  className="button normal secondary"
>
  <span>
    Hello
  </span>
</button>
`;

exports[`Card can have highlight with color color success 1`] = `
<button
  className="button normal success"
>
  <span>
    Hello
  </span>
</button>
`;

exports[`Card renders a cardbody 1`] = `
<Card>
  <div
    className="card"
  >
    <CardBody>
      <div
        className="card-body"
      >
        Hello
      </div>
    </CardBody>
  </div>
</Card>
`;

exports[`Card renders cardbuttons 1`] = `
<Card>
  <div
    className="card"
  >
    <CardButtons>
      <div
        className="card-buttons"
      >
        <Button
          arrow={false}
          color="default"
          text="Hello"
          type="normal"
        >
          <button
            className="button normal default"
          >
            <span>
              Hello
            </span>
          </button>
        </Button>
      </div>
    </CardButtons>
  </div>
</Card>
`;

exports[`Card renders correctly 1`] = `
<div
  className="card"
/>
`;

exports[`Card renders subtitle 1`] = `
<Card>
  <div
    className="card"
  >
    <CardSubtitle>
      <div
        className="card-subtitle"
      >
        Hello
      </div>
    </CardSubtitle>
  </div>
</Card>
`;

exports[`Card renders title 1`] = `
<Card>
  <div
    className="card"
  >
    <CardTitle>
      <div
        className="card-title"
      >
        Hello
      </div>
    </CardTitle>
  </div>
</Card>
`;
